src/preferences/dialog/dlgpreferences.cpp

```diff
@@ -421,8 +421,16 @@ void DlgPreferences::slotButtonPressed(QAbstractButton* pButton) {
         emit applyPreferences();
         break;
     case QDialogButtonBox::AcceptRole:
-        // Same as Apply but close the dialog
         emit applyPreferences();
+        // Check if all pages are in a valid state before accepting
+        for (const PreferencesPage& page : std::as_const(m_allPages)) {
+            if (page.pDlg && !page.pDlg->okayToClose()) {
+                // If any page is not okay to close, switch to it and don't accept
+                switchToPage(page.pTreeItem->text(0), page.pDlg);
```

Comment:
Let's use
```suggestion
                contentsTreeWidget->setCurrentItem(page.pTreeItem);
```
See my explanation in the comment on the changes in `switchToPage()`